clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -614,6 +615,24 @@ public static String stackTrace(Throwable e) {
         return Utils.readBytes(buffer, 0, buffer.limit());
     }
 
+    /**
+     * Read a file as string and return the content. Unlike {@link Utils#readFileAsString(String)}, the file
+     * will be treated as a stream and a character is read at a time. This allows the program to read from say
+     * a FIFO (opened with S_IFIFO flag), where size of file is not known in advance.
+     */
+    public static String readFileStreamAsString(String path) {
+        StringBuilder fileText = new StringBuilder();
+        try (FileInputStream fileInputStream = new FileInputStream(path)) {
+            int nextChar;
+            while ((nextChar = fileInputStream.read()) != -1) {
+                fileText.append((char) nextChar);
+            }
+        } catch (Exception ex) {
```

Comment:
I think this can only catch `IOException`s. We can be more explicit.

Reply:
Done.